tornado-api/src/main/java/uk/ac/manchester/tornado/api/types/tensors/DType.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2024, APT Group, Department of Computer Science,
+ * The University of Manchester.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.manchester.tornado.api.types.tensors;
+
+import java.lang.foreign.ValueLayout;
+
+/**
+ * The {@code DType} enum represents the various data types supported by a data management or processing system.
+ * Each enum constant specifies a data type, its size in bytes, and its corresponding {@link ValueLayout} for memory layout.
+ * This allows for easy management and interpretation of different types of data in a uniform way.
+ */
+public enum DType {
```

Comment:
From the Java side, how do you control is signed or unsigned? How unsigned data types are used? Do you have any example?